bundles/org.openhab.binding.ferroamp/src/main/java/org/openhab/binding/ferroamp/internal/FerroampMqttCommunication.java

```diff
@@ -0,0 +1,1028 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.ferroamp.internal;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.ferroamp.dto.GetGeneralLx;
+import org.openhab.binding.ferroamp.dto.GetGeneralValues;
+import org.openhab.binding.ferroamp.dto.GetUdc;
+import org.openhab.core.io.transport.mqtt.MqttBrokerConnection;
+import org.openhab.core.io.transport.mqtt.MqttMessageSubscriber;
+import org.openhab.core.thing.Thing;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+
+/**
+ * The {@link FerroampMqttCommunication} is responsible for communication with the Ferroamp-systems Mqtt-broker.
+ *
+ * @author Örjan Backsell - Initial contribution
+ *
+ */
+
+@NonNullByDefault
+public class FerroampMqttCommunication implements MqttMessageSubscriber {
+    static String[] ehubChannelsUpdateValues = new String[0];
+    static String[] ssoS1ChannelsUpdateValues = new String[0];
+    static String[] ssoS2ChannelsUpdateValues = new String[0];
+    static String[] ssoS3ChannelsUpdateValues = new String[0];
+    static String[] ssoS4ChannelsUpdateValues = new String[0];
+    static String[] esoChannelsUpdateValues = new String[0];
+    static String[] esmChannelsUpdateValues = new String[0];
+
+    static boolean isSsoChecked = false;
+    static String ssoS1IdCheck = "";
+    static String ssoS2IdCheck = "";
+    static String ssoS3IdCheck = "";
+    static String ssoS4IdCheck = "";
+
+    private final static Logger logger = LoggerFactory.getLogger(FerroampMqttCommunication.class);
+
+    public FerroampMqttCommunication(Thing thing) {
+        super();
+    }
+
+    // Handles request topic
+    static void sendPublishedTopic(String payload, FerroampConfiguration ferroampConfig) {
+        MqttBrokerConnection localConfigurationConnection = FerroampHandler.getFerroampConnection();
+        Objects.requireNonNull(localConfigurationConnection,
+                "MqttBrokerConnection localConfigurationConnection cannot be null");
+        localConfigurationConnection.start();
+        localConfigurationConnection.setCredentials(ferroampConfig.userName, ferroampConfig.password);
+        localConfigurationConnection.publish(FerroampBindingConstants.REQUEST_TOPIC, payload.getBytes(), 1, false);
+    }
+
+    // Handles respective topic type
+    void getSubscribedTopic(String topic, FerroampConfiguration ferroampConfig) {
+        MqttBrokerConnection localSubscribeConnection = FerroampHandler.getFerroampConnection();
+        Objects.requireNonNull(localSubscribeConnection,
+                "MqttBrokerConnection localSubscribeConnection cannot be null");
+        if ("ehubTopic".equals(topic)) {
```

Comment:
Could you extract these topics (magix strings) to constants? I think these are used in multiple files, if so they should be avaialble in the bindingconstants file otherwise at class level.

Reply:
20250319
Changed to bindingconstants, Please see line 71 - 103. I have made the change for each topic